### PR TITLE
Azure - Add policy assignments resource

### PR DESCRIPTION
**Pull request by logachev**
1. Policy assignment resource
2. Updated docs
3. Tests - pending, can't deploy assignment via CLI. Stay tuned.

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/entry.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
 import c7n_azure.resources.vmss
 import c7n_azure.resources.batch
 import c7n_azure.resources.appserviceplan  # noqa: F401
+import c7n_azure.resources.policy_assignments
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/entry.py]: I believe we put `# noqa: F401` on the last import

File: tools/c7n_azure/tests/templates/provision.sh [hunk 2]
```diff
@@ -29,3 +29,11 @@ if [ $# -eq 0 ] || [[ "$@" =~ "containerservice" ]]; then
 else
   echo "Skipping $rgName"
 fi
+
+# Deploy Azure Policy Assignment
+if [ $# -eq 0 ] || [[ "$@" =~ "policyassignment" ]]; then
+  az policy assignment create --display-name cctestpolicy --name cctestpolicy --policy '06a78e20-9358-41c9-923c-fb736d382a4d'
```
Comment (aluong) [inline tools/c7n_azure/tests/templates/provision.sh]: Where does `06a78e20-9358-41c9-923c-fb736d382a4d` come from? A comment may be helpful.

File: tools/c7n_azure/c7n_azure/resources/policy_assignments.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c7n_azure.provider import resources
+from c7n_azure.resources.arm import ArmResourceManager
+
+
+@resources.register('policyassignments')
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/resources/policy_assignments.py]: Can we add another work item to create the policy initiative?

Comment (logachev) [reply tools/c7n_azure/c7n_azure/resources/policy_assignments.py]: Initiative and definition is the same from policy assignment perspective.

**Pull request merged**